tfhe/src/integer/server_key/mod.rs

```diff
@@ -33,13 +33,17 @@ impl From<ServerKey> for crate::shortint::ServerKey {
     }
 }
 
-/// Compute the [`MaxDegree`] for an integer server key (compressed or uncompressed). This formula
-/// provisions a free carry bit. This allows carry propagation between shortint blocks in a
-/// [`RadixCiphertext`](`crate::integer::RadixCiphertext`), as that process requires adding a bit of
-/// carry from one shortint block to the next, which would overflow and lead to wrong results if we
-/// did not provision that carry bit.
+/// Compute the [`MaxDegree`] for an integer server key (compressed or uncompressed).
+/// To allow carry propagation between shortint blocks in a
+/// [`RadixCiphertext`](`crate::integer::RadixCiphertext`) (which includes adding the extracted
+/// carry from one shortint block to the next block), this formula provisions space to add a carry.
 fn integer_server_key_max_degree(parameters: crate::shortint::ShortintParameterSet) -> MaxDegree {
-    MaxDegree((parameters.message_modulus().0 - 1) * parameters.carry_modulus().0 - 1)
+    let full_max_degree = parameters.message_modulus().0 * parameters.carry_modulus().0 - 1;
+
+    let carry_max_degree = parameters.carry_modulus().0 - 1;
```

Comment:
I think this is not correct, the value to consider here is `parameters.carry_modulus().0 - 1` while for most parameters sets the two moduli are equal.

@tmontaigu to confirm

Reply:
carry modulus makes more sense actually, apologies, I was focused on the formula we wrote in the issue, all good !